app/controllers/news_controller.rb

```diff
@@ -6,6 +6,11 @@ class NewsController < ApplicationController
 
   def index
     @articles = NewsArticle.published.by_release_date.by_published_at
+
+    if params[:section].present?
+      set_section
+      @articles = @articles.by_section(@section)
```

Comment:
I believe scopes are lazily applied, and the `by_section` scope is just added on to the already-existing results. I could be wrong on this.